hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUpdateNftsHandler.java

```diff
@@ -52,71 +54,58 @@
  */
 @Singleton
 public class TokenUpdateNftsHandler implements TransactionHandler {
-
-    private final TokenUpdateNftValidator tokenUpdateNftValidator;
+    private final TokenAttributesValidator validator;
 
     @Inject
-    public TokenUpdateNftsHandler(@NonNull final TokenUpdateNftValidator tokenUpdateNftValidator) {
-        this.tokenUpdateNftValidator = tokenUpdateNftValidator;
+    public TokenUpdateNftsHandler(@NonNull final TokenAttributesValidator validator) {
+        this.validator = validator;
     }
 
     @Override
     public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
         requireNonNull(txn);
         final var op = txn.tokenUpdateNftsOrThrow();
         validateTruePreCheck(op.hasToken(), INVALID_TOKEN_ID);
+        validateTrue(!op.serialNumbers().isEmpty(), MISSING_SERIAL_NUMBERS);
     }
 
     @Override
     public void preHandle(@NonNull final PreHandleContext context) throws PreCheckException {
         requireNonNull(context);
-        final var op = context.body().tokenUpdateNftsOrThrow();
         pureChecks(context.body());
-        final var tokenId = op.tokenOrThrow();
-        final var tokenStore = context.createStore(ReadableTokenStore.class);
-        final var tokenMetadata = tokenStore.getTokenMeta(tokenId);
-        if (tokenMetadata == null) throw new PreCheckException(INVALID_TOKEN_ID);
     }
 
     @Override
     public void handle(@NonNull HandleContext context) throws HandleException {
         requireNonNull(context);
         final var txnBody = context.body();
         final var op = txnBody.tokenUpdateNftsOrThrow();
-        final var tokenNftId = op.tokenOrThrow();
+        final var token = op.tokenOrThrow();
 
-        final var validationResult = tokenUpdateNftValidator.validateSemantics(context, op);
-        final var token = validationResult.token();
+        validateSemantics(context, op);
         final var nftStore = context.writableStore(WritableNftStore.class);
 
         // Wrap in Set to de-duplicate serial numbers
-        final var nftSerialNums = new ArrayList<>(new LinkedHashSet<>(op.serialNumbers()));
-        var nftCount = nftSerialNums.size();
-        validateTrue(nftCount <= nftStore.sizeOfState(), INVALID_TRANSACTION_BODY);
-        validateTrue(!nftSerialNums.isEmpty(), INVALID_TRANSACTION_BODY);
-        final var tokensConfig = context.configuration().getConfigData(TokensConfig.class);
-        final var nftsAreEnabled = tokensConfig.nftsAreEnabled();
-        if (nftCount > 1) {
-            validateTrue(nftsAreEnabled, NOT_SUPPORTED);
-        }
-        updateNftMetadata(nftSerialNums, nftStore, tokenNftId, op);
+        final var nftSerialNums = new LinkedHashSet<>(op.serialNumbers());
+        validateTrue(nftSerialNums.size() <= nftStore.sizeOfState(), INVALID_NFT_ID);
```

Comment:
Should this be MISSING_SERIAL_NUMBERS ?

Reply:
No, I think the serials provided are invalid if they are less what are in state. I used `MISSING_SERIAL_NUMBERS` when there are no serial numbers provided in transaction body. I am still not sure if we should fail there. But since we are charging based on serial numbers, I thought we should.